docs/vm-configuration.md

```diff
@@ -264,3 +264,312 @@ the generation of multiple libvirt XML documents, the domain XML and associated
+        </disk>
+        <interface type="bridge">
+            <address bus="0x00" domain="0x0000" function="0x0" slot="0x03" type="pci"/>
+            <mac address="00:aa:bb:cc:dd:ee"/>
+            <model type="virtio"/>
+            <source bridge="mgmtnet"/>
+            <filterref filter="no-mac-spoofing"/>
+        </interface>
+        <graphics autoport="yes" passwd="*****" passwdValidTo="1970-01-01T00:00:01" port="-1" tlsPort="-1" type="spice">
+            <listen network="mgmtnet" type="network"/>
+        </graphics>
+    </devices>
+    <os>
+        <type arch="x86_64" machine="pc-i440fx-rhel7.3.0">hvm</type>
+        <smbios mode="sysinfo"/>
+    </os>
+</domain>
+```
+
+roughly translates to the following yaml (not everything is implemented,
+therefore few placeholders are used):
+
+```yaml
+apiVersion: kubevirt.io/v1alpha1
+kind: VM
+metadata:
+  name: VM
+spec:
+  domain:
+    memory:
+      size: 4096
+      current: 4096
+    maxmemory:
+      size: 16777216
+      slots: 16
+    iothreads: 1
+    vcpu:
+      count: 16
+      current: 4
+    sysinfo:
+      manufacturer: KubeVirt
+      product: KubeVirt Node
+      serial: 11111111-FEED-DEAD-BEEF-111111111111
+      uuid: 11111111-FEED-DEAD-BEEF-111111111111
+    clock:
+      - timer: hypervclock
+        present: true
+      - timer: rtc
+        tickPolicy: catchup
+      - timer: pit
+        tickPolicy: delay
+      - timer: hpet
+        present: false
+    features:
+      - feature: acpi
+      - feature: hyperv
+        relaxed: true
+        vapic: true
+        spinlocks:
+          retries: 8191
+          state: true
+    cpu:
+      match: exact
+      model: Westmere
+      topology:
+        cores: 1
+        sockets: 16
+        threads: 1
+      numa:
+        - cell:
+          cpus: [1, 2, 3, 4]
+          memory: 4194304
+    devices:
+     channels:
+       - type: unix
+         source:
+           mode: bind
+           path: /var/lib/libvirt/qemu/channels/11111111-feed-dead-beef-111111111111.org.qemu.guest_agent.0
+         target:
+           name: org.qemu.guest_agent.0
+           type: virtio
+       - type: spicevmc
+         target:
+           name: com.redhat.spice.0
+           type: virtio
+     memballoon:
+       model: virtio
+     rng:
+       backend: random
+       path: /dev/urandom
+     controllers:
+       - type: scsi
+         model: virtio-scsi
+      video:
+      - type: qxl
+        ram: 65536
+        vgamem: 16384
+        vram: 8192
+      disks:
+      - device: cdrom
+        snapshot: false
+        type: file
+        source:
+          startupPolicy: optional
+        target:
+          bus: ide
+          dev: hdc
+        readonly: true
+      - device: disk 
+        snapshot: false
+        type: file
+        source:
+          file: /mntpoint/11111111-1111-1111-1111-111111111111/images/22222222-2222-2222-2222-222222222222/22222222-2222-2222-2222-222222222222
```

Comment:
I think that even in the "lower level" representation, the VM configuration should not point to the the exact file of the disk. Let's say that we do live storage migration then we initially create a snapshot on top of each disk - such representation would make us change the VM configuration at this point and later on when merging the temporary snapshot. I think that a better way is pointing to the disk (like in line 546), assuming that the leaf volume is used.